charms/istio-pilot/src/istio_gateway_name_provider.py

```diff
@@ -0,0 +1,21 @@
+import logging
+from ops.framework import Object
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_RELATION_NAME = "gateway"
+
+
+class GatewayProvider(Object):
+    def __init__(self, charm, relation_name=DEFAULT_RELATION_NAME):
+        super().__init__(charm, relation_name)
+
+    def send_gateway_relation_data(self, charm, gateway_name, gateway_namespace):
+        relations = self.model.relations["gateway"]
+        for relation in relations:
+            relation.data[charm].update(
+                {
+                    "gateway_name": gateway_name,
+                    "gateway_namespace": gateway_namespace,
+                }
+            )
```

Comment:
Why was the leadership check removed entirely? As I understand, this part of the code will fail when a non-leader executes it.

Reply:
There is a leadership check in operator init